core-rust-bridge/src/main/java/com/radixdlt/environment/StateHashTreeGcConfig.java

```diff
@@ -0,0 +1,85 @@
+ * Work or any use of the Work complies with any law or regulation in any territory where
+ * it may be implemented or used or that it will be appropriate for any specific purpose.
+ *
+ * Neither the licensor nor any current or former employees, officers, directors, partners,
+ * trustees, representatives, agents, advisors, contractors, or volunteers of the Licensor
+ * shall be liable for any direct or indirect, special, incidental, consequential or other
+ * losses of any kind, in tort, contract or otherwise (including but not limited to loss
+ * of revenue, income or profits, or loss of use or data, or loss of reputation, or loss
+ * of any economic or other opportunity of whatsoever nature or howsoever arising), arising
+ * out of or in connection with (without limitation of any use, misuse, of any ledger system
+ * or use made or its functionality or any performance or operation of any code or protocol
+ * caused by bugs or programming or logic errors or otherwise);
+ *
+ * A. any offer, purchase, holding, use, sale, exchange or transmission of any
+ * cryptographic keys, tokens or assets created, exchanged, stored or arising from any
+ * interaction with the Work;
+ *
+ * B. any failure in a transmission or loss of any token or assets keys or other digital
+ * artefacts due to errors in transmission;
+ *
+ * C. bugs, hacks, logic errors or faults in the Work or any communication;
+ *
+ * D. system software or apparatus including but not limited to losses caused by errors
+ * in holding or transmitting tokens by any third-party;
+ *
+ * E. breaches or failure of security including hacker attacks, loss or disclosure of
+ * password, loss of private key, unauthorised use or misuse of such passwords or keys;
+ *
+ * F. any losses including loss of anticipated savings or other benefits resulting from
+ * use of the Work or any changes to the Work (however implemented).
+ *
+ * You are solely responsible for; testing, validating and evaluation of all operation
+ * logic, functionality, security and appropriateness of using the Work for any commercial
+ * or non-commercial purpose and for any reproduction or redistribution by You of the
+ * Work. You assume all risks associated with Your use of the Work and the exercise of
+ * permissions under this License.
+ */
+
+package com.radixdlt.environment;
+
+import com.radixdlt.sbor.codec.CodecMap;
+import com.radixdlt.sbor.codec.StructCodec;
+import com.radixdlt.utils.UInt32;
+import com.radixdlt.utils.UInt64;
+
+public record StateHashTreeGcConfig(
+    UInt32 intervalSec, UInt64 stateVersionHistoryLength, UInt64 maxDbLockingDurationMillis) {
+  public static void registerCodec(CodecMap codecMap) {
+    codecMap.register(
+        StateHashTreeGcConfig.class,
+        codecs -> StructCodec.fromRecordComponents(StateHashTreeGcConfig.class, codecs));
+  }
+
+  public static StateHashTreeGcConfig forTesting() {
+    // Effectively "no GC during tests":
```

Comment:
why no GC in tests?

Reply:
I have not implemented the required `StateHashTreeGcStore` trait for MemoryDB and it would probably fail many tests anyway :(

But in the meantime, looks like we are getting rid of entire MemoryDB (https://github.com/radixdlt/babylon-node/pull/716/commits/a3509d6b5eca79dab51939c62f31448293f3da0a), so I should soon be able to adjust this 👍

Reply:
✅